pkg/diagnostics/metrics.go

```diff
@@ -0,0 +1,33 @@
+package diagnostics
+
+import (
+	"github.com/dapr/components-contrib/middleware/http/nethttpadaptor"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	http_metrics "github.com/improbable-eng/go-httpwares/metrics"
+	http_prometheus "github.com/improbable-eng/go-httpwares/metrics/prometheus"
+	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/fasthttpadaptor"
+	grpc_go "google.golang.org/grpc"
+)
+
+// MetricsGRPCMiddlewareStream gets a metrics enabled GRPC stream middlware
+func MetricsGRPCMiddlewareStream() grpc_go.StreamServerInterceptor {
+	return grpc_prometheus.StreamServerInterceptor
+}
+
+// MetricsGRPCMiddlewareUnary gets a metrics enabled GRPC unary middlware
+func MetricsGRPCMiddlewareUnary() grpc_go.UnaryServerInterceptor {
+	return grpc_prometheus.UnaryServerInterceptor
+}
+
+// MetricsHTTPMiddleware gets a metrics enabled HTTP middleware
+func MetricsHTTPMiddleware(next fasthttp.RequestHandler) fasthttp.RequestHandler {
+	// TODO: support custom labels
+	mw := http_metrics.Middleware(http_prometheus.ServerMetrics(
```

Comment:
is there a way to add prefix(`dapr`) to metric label?

Reply:
investigating this - custom label and namespace prefixes don't seem to be exposed easily but I might find a way

Reply:
I think the simplest way might be to add the labels on the prometheus scraping config, you can also use `metric_relabel_configs` to change the metrics names with a prefix.

Reply:
@jjcollinge thanks for looking for it. If we can add aggregation dimension(properties) to metric, it would be the another option. But I agree that we would not need it if we do not have simple solution.